pkg/sysregistriesv2/system_registries_v2.go

```diff
@@ -338,55 +338,81 @@ func (config *V2RegistriesConf) postProcess() error {
-	}
-
-	userRegistriesDirPath := filepath.Join(homedir.Get(), userRegistriesDir)
-	if _, err := os.Stat(userRegistriesDirPath); err == nil {
-		return userRegistriesDirPath
-	}
-
-	if ctx != nil && ctx.RootForImplicitAbsolutePaths != "" {
-		return filepath.Join(ctx.RootForImplicitAbsolutePaths, systemRegistriesConfDirPath)
-	}
-
-	return systemRegistriesConfDirPath
+	return newConfigWrapper(ctx).configDirPath
 }
 
 // configWrapper is used to store the paths from ConfigPath and ConfigDirPath
 // and acts as a key to the internal cache.
 type configWrapper struct {
-	configPath    string
+	// path to the registries.conf file
+	configPath string
+	// path to system-wide registries.conf.d directory
 	configDirPath string
+	// path to user specificed registries.conf.d directory
+	userConfigDirPath string
 }
 
 // newConfigWrapper returns a configWrapper for the specified SystemContext.
 func newConfigWrapper(ctx *types.SystemContext) configWrapper {
-	return configWrapper{
-		configPath:    ConfigPath(ctx),
-		configDirPath: ConfigDirPath(ctx),
+	var wrapper configWrapper
+	if ctx == nil {
+		wrapper.configPath = systemRegistriesConfPath
+		wrapper.configDirPath = systemRegistriesConfDirPath
+		wrapper.userConfigDirPath = getUserConfigDirPath(ctx)
+		return wrapper
 	}
+	userRegistriesFilePath := filepath.Join(homedir.Get(), userRegistriesFile)
+	if ctx.SystemRegistriesConfPath != "" {
+		wrapper.configPath = ctx.SystemRegistriesConfPath
+		wrapper.configDirPath = getConfigDirPath(ctx)
+		wrapper.userConfigDirPath = getUserConfigDirPath(ctx)
+		return wrapper
+	}
+	if _, err := os.Stat(userRegistriesFilePath); err == nil {
+		wrapper.configPath = userRegistriesFilePath
+		wrapper.userConfigDirPath = getUserConfigDirPath(ctx)
+		return wrapper
+	}
+	wrapper.configPath = filepath.Join(ctx.RootForImplicitAbsolutePaths, systemRegistriesConfPath)
+	wrapper.configDirPath = getConfigDirPath(ctx)
+	wrapper.userConfigDirPath = getUserConfigDirPath(ctx)
+	return wrapper
+}
+
+// getUserConfigDirPath returns user specified path of registries.conf.d if exists, returns "" otherwise
+func getUserConfigDirPath(ctx *types.SystemContext) string {
+	if ctx != nil && ctx.SystemRegistriesConfDirPath != "" {
+		return ctx.SystemRegistriesConfDirPath
+	}
+	userRegistriesDirPath := filepath.Join(homedir.Get(), userRegistriesDir)
+	if _, err := os.Stat(userRegistriesDirPath); err == nil {
+		return userRegistriesDirPath
+	}
+	return ""
+}
+
+func getConfigDirPath(ctx *types.SystemContext) string {
+	if ctx == nil {
+		return systemRegistriesConfDirPath
+	}
+	return filepath.Join(ctx.RootForImplicitAbsolutePaths, systemRegistriesConfDirPath)
+}
+
+// ConfigurationSourceDescription returns a string containres paths of registries.conf and registries.conf.d
+func ConfigurationSourceDescription(ctx *types.SystemContext) string {
+	wrapper := newConfigWrapper(ctx)
+	return strings.Join([]string{wrapper.configPath, wrapper.configDirPath}, ",")
```

Comment:
(Absolutely non-blocking: Maybe
```suggestion
	return strings.Join([]string{wrapper.configPath, wrapper.configDirPath}, ", ")
```
to make this easier to read (and marginally harder to parse by machines 👿 ))

Reply:
fixed in the refactoring commit